src/python/pants/jvm/test/junit_test.py

```diff
@@ -633,6 +633,7 @@ def run_junit_test(
     env: Mapping[str, str] | None = None,
 ) -> TestResult:
     args = [
+        "--no-jvm-enable-lockfile-targets",
```

Comment:
Is there a way to avoid needing to disable this feature in ~all the tests? The effect of them needing to be disabled seems to be 1) more boilerplate, 2) less test coverage.

Reply:
I guess going the other way would mean we need to add the new rules instead. Perhaps preferable?

Reply:
Unless they were bundled in with the inference `@rules` perhaps? But I'm not sure what disabling this is intended to skip. If just not specifying the flag and not installing the `@rules` causes no side-effect on the test, then IMO would just remove the flag.

Reply:
The tests fail without it, otherwise I wouldn't have bothered adding this everywhere 😂 
So yea, maybe bundling somewhere so we get them for free without having to touch the tests would be the right thing to do..